File: TDDMicroExercises/Ruby/tire_pressure_monitoring_system/alarm.rb
require_relative './sensor'

class Alarm

  attr_reader :alarm_on

  def initialize
    @sensor = Sensor.new
    @alarm_on = false
  end

  def check
    pressure = @sensor.pop_next_pressure_psi_value()

    @alarm_on = true if pressure < LOW_PRESSURE || HIGH_PRESSURE < pressure
    end

private

  LOW_PRESSURE = 17
  HIGH_PRESSURE = 21
end

File: TDDMicroExercises.YoursSolutions/Java-viswans/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm
{
    public static final double LowPressureThreshold = 17;
    public static final double HighPressureThreshold = 21;

    private ISensor sensor;	
    private Range validPressureRange;

    private boolean alarmOn = false;
    
    public Alarm(ISensor sensor, Range validPressureRange) {
    	this.sensor = sensor;
    	this.validPressureRange = validPressureRange;
	}

    public void check()
    {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (!validPressureRange.inRange(psiPressureValue))
        {
            alarmOn = true;
        }
    }

    public boolean isAlarmOn()
    {
        return alarmOn; 
    }
    
}
